src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -231,11 +231,16 @@ private async void ProcessIncomingFrames()
             }
         }
 
-        // Note, this will return null for a streamId that's not in use.
+        // Note, this will return null for a streamId that's no longer in use.
         // Callers must check for this and send a RST_STREAM or ignore as appropriate.
+        // If the streamId is invalid or the stream is idle, calling this function
+        // will result in a connection level error.
         private Http2Stream GetStream(int streamId)
         {
-            if (streamId <= 0)
+            // If we implement support for Push Promise, this will need to be updated to
+            // track the highest stream ID used by the server in addition to the highest
+            // ID used by the client.
+            if (streamId <= 0 || streamId >= _nextStream)
```

Comment:
Do we have an issue for Push Promise?  We should link a TODO here.

Reply:
We don't currently have an issue tracking support for Push Promise. I'll go ahead and open one targeting future, since we still haven't decided if we're going to implement Push Promise or how we would expose it.

Reply:
Sounds good, I think it's easier to search in the code later if we can link a issue/TODO.

Reply:
I've opened #34192 to track implementing (or considering implementing) Push Promise. I'll add a note to the code.